packages/sfui/frameworks/vue/components/SfScrollable/SfScrollable.vue

```diff
@@ -0,0 +1,141 @@
+  activeIndex: {
+    type: Number,
+    default: undefined,
+  },
+  reduceMotion: {
+    type: Boolean,
+    default: undefined,
+  },
+  drag: {
+    type: Boolean,
+    default: undefined,
+  },
+});
+const emit = defineEmits<{
+  (e: 'onDragChange', data: SfScrollableOnDragChangeData): void;
+  (e: 'onScroll', data: SfScrollableOnScrollData): void;
+  (e: 'onPrev'): void;
+  (e: 'onNext'): void;
+}>();
+const { direction, activeIndex, reduceMotion, drag } = toRefs(props);
+
+const { getContainerRef, state, getNextButtonProps, getPrevButtonProps } = useScrollable(
+  computed(() => ({
+    ...reactive({
+      direction,
+      activeIndex,
+      reduceMotion,
+      drag,
+    }),
+    onDragChange: (data) => emit('onDragChange', data),
+    onScroll: (data) => emit('onScroll', data),
+    onPrev: () => emit('onPrev'),
+    onNext: () => emit('onPrev'),
+  })),
+);
+
+const isHorizontal = computed(() => props.direction === SfScrollableDirection.horizontal);
+</script>
+
+<template>
+  <div :class="['items-center', 'relative', isHorizontal ? 'flex' : 'flex-col h-full inline-flex', wrapperClass]">
+    <div v-if="$slots.previouButton" v-bind="getPrevButtonProps"><slot name="previouButton" /></div>
+    <SfButton
+      v-else-if="buttonsPlacement === SfScrollableButtonsPlacement.blocked"
+      variant="secondary"
+      size="lg"
+      square
+      :class="['!rounded-full bg-white !ring-gray-200 !text-gray-500', isHorizontal ? 'mr-4' : 'mb-4 rotate-90']"
+      v-bind="getPrevButtonProps"
+    >
+      <SfIconChevronLeft />
+    </SfButton>
+    <component
+      :is="tag"
+      ref="getContainerRef"
+      :class="[
+        'motion-safe:scroll-smooth',
+        {
+          'overflow-x-auto flex gap-4': isHorizontal,
+          'overflow-y-auto flex flex-col gap-4': !isHorizontal,
+          'cursor-grab': state.isDragged,
+        },
+      ]"
+      v-bind="{ ...$attrs, ...props }"
+    >
+      <div v-if="$slots.previouButton" v-bind="getPrevButtonProps"><slot name="previouButton" /></div>
+      <SfButton
+        v-else-if="buttonsPlacement === SfScrollableButtonsPlacement.floating"
+        variant="secondary"
+        size="lg"
+        square
+        :class="[
+          'absolute !rounded-full bg-white !ring-gray-200 !text-gray-500',
+          isHorizontal ? 'left-4' : 'top-4 rotate-90',
+        ]"
+        v-bind="getPrevButtonProps"
+      >
+        <SfIconChevronLeft />
+      </SfButton>
+      <slot />
+      <div v-if="$slots.nextButton" v-bind="getNextButtonProps"><slot name="nextButton" /></div>
+      <SfButton
+        v-else-if="buttonsPlacement === SfScrollableButtonsPlacement.floating"
+        variant="secondary"
+        size="lg"
+        square
+        :class="[
+          'absolute !rounded-full bg-white !ring-gray-200 !text-gray-500',
+          isHorizontal ? 'right-4' : 'bottom-4 rotate-90',
+        ]"
+        v-bind="getNextButtonProps"
+      >
+        <SfIconChevronRight />
```

Comment:
Button seems to not disable randomly when hitting border
This happens for both Vue/React components.
<img width="283" alt="image" src="https://user-images.githubusercontent.com/32792205/230591755-6dc41e92-5626-4539-a8c1-c2f555381bb2.png">

Reply:
Sure, since `prev` button gets `disabled` state on far left scroll position 
<img width="277" alt="Screenshot 2023-04-07 at 17 01 17" src="https://user-images.githubusercontent.com/32792205/230631495-891edcae-0cfc-4b53-8285-1e7ae73b934c.png">


`next` button always is in state `active` & `hoverable`

Reply:
oh you mean, that visually there are always `disable` style applied even if its not disabled? if thats correct, then thank you for catching that https://github.com/vuestorefront/storefront-ui/pull/2571/commits/d6c212012b50e96c029c7172f433b184ada612c4

Reply:
Seems like this issue still exists, buttons behaviour is inconsistent

https://user-images.githubusercontent.com/32792205/231117605-a29cfc44-f808-4b80-9e44-7739566edfef.mov